reana_client/api/client.py

```diff
@@ -764,3 +763,100 @@ def get_workflow_disk_usage(workflow, parameters, access_token):
         raise Exception(e.response.json()['message'])
     except Exception as e:
         raise e
+
+
+def add_secrets(secrets, overwrite, access_token):
+    """Add a new secret.
+
+    :param secrets: dictionary containing all the secrets to be sent.
+    :param overwrite: the flag telling if should be overwritten when they
+        already exist.
+    :param access_token: access token of the current user.
+
+    """
+    try:
+        (response,
+            http_response) = current_rs_api_client.api.add_secrets(
+            secrets=secrets,
+            access_token=access_token,
+            overwrite=overwrite).result()
+        if http_response.status_code == 201:
+            return response
+        else:
+            raise Exception(
+                "Expected status code 201 but replied with "
+                "{status_code}".format(
+                    status_code=http_response.status_code))
+
+    except HTTPError as e:
+        logging.debug(
+            'Secrets could not be added: '
+            '\nStatus: {}\nReason: {}\n'
+            'Message: {}'.format(e.response.status_code,
+                                 e.response.reason,
+                                 e.response.json()['message']))
+        raise Exception(e.response.json()['message'])
+    except Exception as e:
+        raise e
+
+
+def delete_secrets(secrets, access_token):
+    """Add a new secret.
+
+    :param secrets: List of secrets names to be deleted.
+    :param access_token: access token of the current user.
+
+    """
+    try:
+        (response,
+            http_response) = current_rs_api_client.api.delete_secrets(
+            secrets=secrets,
+            access_token=access_token).result()
+        if http_response.status_code in [200, 404]:
+            return response
+        else:
+            raise Exception(
+                "Expected status code 200 but replied with "
+                "{status_code}".format(
+                    status_code=http_response.status_code))
+
+    except HTTPError as e:
```

Comment:
The `404` will be an HTTPError, so we need to handle the `404` case in the exception body.